h2o-core/src/main/java/water/ExternalFrameWriterClient.java

```diff
@@ -124,77 +125,79 @@ public void finalizeFrame(String keyName, long[] rowsPerChunk, byte[] colTypes,
     waitForRequestToFinish(timeout, requestType);
   }
 
-    public void close() throws IOException, ExternalFrameConfirmationException {
-        waitForRequestToFinish(timeout, ExternalBackendRequestType.WRITE_TO_CHUNK.getByte());
-        channel.close();
-    }
+  public void close() throws IOException, ExternalFrameConfirmationException {
+    // write remaining data
+    writeToChannel(ab, channel);
+    waitForRequestToFinish(timeout, ExternalBackendRequestType.WRITE_TO_CHUNK.getByte());
+    channel.close();
```

Comment:
What if the wait operation times out ? Is the channel.close() is going to be closed and resources freed ?